Makefile

```diff
@@ -84,22 +84,24 @@ po2json:
 
 .PHONY: release
 release:
-	$(SED) -ri 's/_converse.VERSION_NAME = "v[0-9]+\.[0-9]+\.[0-9]+";/ _converse.VERSION_NAME = "v$(VERSION)";/' src/headless/converse-core.js
-	$(SED) -ri s/Version:\ [0-9]\+\.[0-9]\+\.[0-9]\+/Version:\ $(VERSION)/ COPYRIGHT
-	$(SED) -ri s/Project-Id-Version:\ Converse\.js\ [0-9]\+\.[0-9]\+\.[0-9]\+/Project-Id-Version:\ Converse.js\ $(VERSION)/ locale/converse.pot
-	$(SED) -ri s/\"version\":\ \"[0-9]\+\.[0-9]\+\.[0-9]\+\"/\"version\":\ \"$(VERSION)\"/ package.json
-	$(SED) -ri s/\"version\":\ \"[0-9]\+\.[0-9]\+\.[0-9]\+\"/\"version\":\ \"$(VERSION)\"/ src/headless/package.json
-	$(SED) -ri s/--package-version=[0-9]\+\.[0-9]\+\.[0-9]\+/--package-version=$(VERSION)/ Makefile
-	$(SED) -ri s/v[0-9]\+\.[0-9]\+\.[0-9]\+\.zip/v$(VERSION)\.zip/ index.html
-	$(SED) -ri s/v[0-9]\+\.[0-9]\+\.[0-9]\+\.tar\.gz/v$(VERSION)\.tar\.gz/ index.html
-	$(SED) -ri s/version\ =\ \'[0-9]\+\.[0-9]\+\.[0-9]\+\'/version\ =\ \'$(VERSION)\'/ docs/source/conf.py
-	$(SED) -ri s/release\ =\ \'[0-9]\+\.[0-9]\+\.[0-9]\+\'/release\ =\ \'$(VERSION)\'/ docs/source/conf.py
-	$(SED) -ri "s/(Unreleased)/`date +%Y-%m-%d`/" CHANGES.md
-	$(SED) -ri "s/cdn.conversejs.org\/[0-9]+\.[0-9]+\.[0-9]+/cdn.conversejs.org\/$(VERSION)/" docs/source/quickstart.rst
+	$(SED) -i '/^_converse.VERSION_NAME =/s/=.*/= "v$(VERSION)";/' src/headless/converse-core.js
+	$(SED) -i '/Version:/s/:.*/: $(VERSION)/' COPYRIGHT
```

Comment:
Hi @joudinet, this looks much cleaner thanks, but I don't know the regex syntax. Would you mind explaining how it works?

Looks like you're first matching `Version:`, and then using `s` to designate substitution of `:.*` with `: $(VERSION)`.

But the substitution match `:*` is now much looser than before, or am I misunderstanding how it works?

Reply:
Yes, it first matches lines that match /regexp/ and applies on each of them the command s/from/to/ to replace the portion matched by 'from' with 'to'. This avoids to repeat yourself in the replacement part.
I looked at instances were these sed commands are applied, an I loose the match regexp such that it still matches the correct portion to be replaced without matching any other part of the document. I don't see the point in making the match part over complex. For example, the ' Version:' in the COPYRIGHT replaces the end of the line after the colon by ': $(VERSION)', since there is nothing else present on this line after the version.
Feel free to harden them if you're afraid such commands might match something they shouldn't replace in a future version.

I also removed the -r (i.e. -E or --regex-extended) option whenever it was not needed. For example, it was confusing in the sed command for CHANGES.md. there were parenthesis in the match part but no \1 in the replacement part to insert the matched expression. I think this command wanted to actually match the parenthesis characters. So, I actually harden the match regexp for this sed command.

Reply:
Thanks for the explanation, I think it's good as it is now.